File: include/libqdebug.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://doc.qt.io/qt-6/qdebug.html
pub const qdebug = struct {
    /// New constructs a new QDebug object.
    ///
    /// ``` device: QtC.QIODevice ```
    pub fn New(device: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_new(@ptrCast(device));
    }

    /// New2 constructs a new QDebug object.
    ///
    /// ``` o: QtC.QDebug ```
    pub fn New2(o: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_new2(@ptrCast(o));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator=)
    ///
    /// ``` self: QtC.QDebug, other: QtC.QDebug ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QDebug_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#swap)
    ///
    /// ``` self: QtC.QDebug, other: QtC.QDebug ```
    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QDebug_Swap(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#resetFormat)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn ResetFormat(self: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_ResetFormat(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#space)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn Space(self: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_Space(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#nospace)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn Nospace(self: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_Nospace(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#maybeSpace)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn MaybeSpace(self: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_MaybeSpace(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#verbosity)
    ///
    /// ``` self: QtC.QDebug, verbosityLevel: i32 ```
    pub fn Verbosity(self: ?*anyopaque, verbosityLevel: i32) QtC.QDebug {
        return qtc.QDebug_Verbosity(@ptrCast(self), @intCast(verbosityLevel));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#verbosity)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn Verbosity2(self: ?*anyopaque) i32 {
        return qtc.QDebug_Verbosity2(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#setVerbosity)
    ///
    /// ``` self: QtC.QDebug, verbosityLevel: i32 ```
    pub fn SetVerbosity(self: ?*anyopaque, verbosityLevel: i32) void {
        qtc.QDebug_SetVerbosity(@ptrCast(self), @intCast(verbosityLevel));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#autoInsertSpaces)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn AutoInsertSpaces(self: ?*anyopaque) bool {
        return qtc.QDebug_AutoInsertSpaces(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#setAutoInsertSpaces)
    ///
    /// ``` self: QtC.QDebug, b: bool ```
    pub fn SetAutoInsertSpaces(self: ?*anyopaque, b: bool) void {
        qtc.QDebug_SetAutoInsertSpaces(@ptrCast(self), b);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#quoteStrings)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn QuoteStrings(self: ?*anyopaque) bool {
        return qtc.QDebug_QuoteStrings(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#setQuoteStrings)
    ///
    /// ``` self: QtC.QDebug, b: bool ```
    pub fn SetQuoteStrings(self: ?*anyopaque, b: bool) void {
        qtc.QDebug_SetQuoteStrings(@ptrCast(self), b);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#quote)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn Quote(self: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_Quote(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#noquote)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn Noquote(self: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_Noquote(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#maybeQuote)
    ///
    /// ``` self: QtC.QDebug ```
    pub fn MaybeQuote(self: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_MaybeQuote(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: QtC.QChar ```
    pub fn OperatorShiftLeft(self: ?*anyopaque, t: QtC.QChar) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft(@ptrCast(self), @ptrCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: bool ```
    pub fn OperatorShiftLeftWithBool(self: ?*anyopaque, t: bool) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithBool(@ptrCast(self), t);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: i8 ```
    pub fn OperatorShiftLeftWithChar(self: ?*anyopaque, t: i8) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithChar(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: i16 ```
    pub fn OperatorShiftLeftWithShort(self: ?*anyopaque, t: i16) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithShort(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: u16 ```
    pub fn OperatorShiftLeftWithUnsignedshort(self: ?*anyopaque, t: u16) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithUnsignedshort(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: i32 ```
    pub fn OperatorShiftLeftWithInt(self: ?*anyopaque, t: i32) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithInt(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: u32 ```
    pub fn OperatorShiftLeftWithUnsignedint(self: ?*anyopaque, t: u32) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithUnsignedint(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: i64 ```
    pub fn OperatorShiftLeftWithLong(self: ?*anyopaque, t: i64) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithLong(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: u64 ```
    pub fn OperatorShiftLeftWithUnsignedlong(self: ?*anyopaque, t: u64) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithUnsignedlong(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: i64 ```
    pub fn OperatorShiftLeftWithQint64(self: ?*anyopaque, t: i64) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithQint64(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: u64 ```
    pub fn OperatorShiftLeftWithQuint64(self: ?*anyopaque, t: u64) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithQuint64(@ptrCast(self), @intCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: f32 ```
    pub fn OperatorShiftLeftWithFloat(self: ?*anyopaque, t: f32) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithFloat(@ptrCast(self), @floatCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: f64 ```
    pub fn OperatorShiftLeftWithDouble(self: ?*anyopaque, t: f64) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithDouble(@ptrCast(self), @floatCast(t));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: []const u8 ```
    pub fn OperatorShiftLeft2(self: ?*anyopaque, t: []const u8) QtC.QDebug {
        const t_Cstring = t.ptr;
        return qtc.QDebug_OperatorShiftLeft2(@ptrCast(self), t_Cstring);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: []const u8 ```
    pub fn OperatorShiftLeftWithQString(self: ?*anyopaque, t: []const u8) QtC.QDebug {
        const t_str = qtc.struct_libqt_string{
            .len = t.len,
            .data = t.ptr,
        };
        return qtc.QDebug_OperatorShiftLeftWithQString(@ptrCast(self), t_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: []u8 ```
    pub fn OperatorShiftLeftWithQByteArray(self: ?*anyopaque, t: []u8) QtC.QDebug {
        const t_str = qtc.struct_libqt_string{
            .len = t.len,
            .data = t.ptr,
        };
        return qtc.QDebug_OperatorShiftLeftWithQByteArray(@ptrCast(self), t_str);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: []const u8 ```
    pub fn OperatorShiftLeftWithQByteArrayView(self: ?*anyopaque, t: []const u8) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithQByteArrayView(@ptrCast(self), t.ptr);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#operator<<)
    ///
    /// ``` self: QtC.QDebug, t: ?*anyopaque ```
    pub fn OperatorShiftLeftWithVoid(self: ?*anyopaque, t: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeftWithVoid(@ptrCast(self), t);
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#maybeQuote)
    ///
    /// ``` self: QtC.QDebug, c: i8 ```
    pub fn MaybeQuote1(self: ?*anyopaque, c: i8) QtC.QDebug {
        return qtc.QDebug_MaybeQuote1(@ptrCast(self), @intCast(c));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebug.html#dtor.QDebug)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QDebug ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QDebug_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qdebugstatesaver.html
pub const qdebugstatesaver = struct {
    /// New constructs a new QDebugStateSaver object.
    ///
    /// ``` dbg: QtC.QDebug ```
    pub fn New(dbg: ?*anyopaque) QtC.QDebugStateSaver {
        return qtc.QDebugStateSaver_new(@ptrCast(dbg));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qdebugstatesaver.html#dtor.QDebugStateSaver)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QDebugStateSaver ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QDebugStateSaver_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qnodebug.html
pub const qnodebug = struct {
    /// New constructs a new QNoDebug object.
    ///
    /// ``` other: QtC.QNoDebug ```
    pub fn New(other: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_new(@ptrCast(other));
    }

    /// New2 constructs a new QNoDebug object and invalidates the source QNoDebug object.
    ///
    /// ``` other: QtC.QNoDebug ```
    pub fn New2(other: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_new2(@ptrCast(other));
    }

    /// CopyAssign shallow copies `other` into `self`.
    ///
    /// ``` self: QtC.QNoDebug, other: QtC.QNoDebug ```
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNoDebug_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    ///
    /// ``` self: QtC.QNoDebug, other: QtC.QNoDebug ```
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNoDebug_MoveAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#space)
    ///
    /// ``` self: QtC.QNoDebug ```
    pub fn Space(self: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_Space(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#nospace)
    ///
    /// ``` self: QtC.QNoDebug ```
    pub fn Nospace(self: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_Nospace(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#maybeSpace)
    ///
    /// ``` self: QtC.QNoDebug ```
    pub fn MaybeSpace(self: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_MaybeSpace(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#quote)
    ///
    /// ``` self: QtC.QNoDebug ```
    pub fn Quote(self: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_Quote(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#noquote)
    ///
    /// ``` self: QtC.QNoDebug ```
    pub fn Noquote(self: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_Noquote(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#maybeQuote)
    ///
    /// ``` self: QtC.QNoDebug ```
    pub fn MaybeQuote(self: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_MaybeQuote(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#verbosity)
    ///
    /// ``` self: QtC.QNoDebug, param1: i32 ```
    pub fn Verbosity(self: ?*anyopaque, param1: i32) QtC.QNoDebug {
        return qtc.QNoDebug_Verbosity(@ptrCast(self), @intCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#maybeQuote)
    ///
    /// ``` self: QtC.QNoDebug, param1: i8 ```
    pub fn MaybeQuote1(self: ?*anyopaque, param1: i8) QtC.QNoDebug {
        return qtc.QNoDebug_MaybeQuote1(@ptrCast(self), @intCast(param1));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qnodebug.html#dtor.QNoDebug)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.QNoDebug ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QNoDebug_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qdebug.html#types
pub const enums = struct {
    pub const VerbosityLevel = enum {
        pub const MinimumVerbosity: i32 = 0;
        pub const DefaultVerbosity: i32 = 2;
        pub const MaximumVerbosity: i32 = 7;
    };
};

File: src_original/libqdebug.cpp
#include <QByteArray>
#include <QByteArrayView>
#include <QChar>
#include <QDebug>
#include <QDebugStateSaver>
#include <QIODevice>
#include <QIODeviceBase>
#include <QNoDebug>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qdebug.h>
#include "libqdebug.h"
#include "libqdebug.hxx"

QDebug* QDebug_new(QIODevice* device) {
    return new QDebug(device);
}

QDebug* QDebug_new2(const QDebug* o) {
    return new QDebug(*o);
}

void QDebug_OperatorAssign(QDebug* self, const QDebug* other) {
    self->operator=(*other);
}

void QDebug_Swap(QDebug* self, QDebug* other) {
    self->swap(*other);
}

QDebug* QDebug_ResetFormat(QDebug* self) {
    QDebug& _ret = self->resetFormat();
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_Space(QDebug* self) {
    QDebug& _ret = self->space();
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_Nospace(QDebug* self) {
    QDebug& _ret = self->nospace();
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_MaybeSpace(QDebug* self) {
    QDebug& _ret = self->maybeSpace();
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_Verbosity(QDebug* self, int verbosityLevel) {
    QDebug& _ret = self->verbosity(static_cast<int>(verbosityLevel));
    // Cast returned reference into pointer
    return &_ret;
}

int QDebug_Verbosity2(const QDebug* self) {
    return self->verbosity();
}

void QDebug_SetVerbosity(QDebug* self, int verbosityLevel) {
    self->setVerbosity(static_cast<int>(verbosityLevel));
}

bool QDebug_AutoInsertSpaces(const QDebug* self) {
    return self->autoInsertSpaces();
}

void QDebug_SetAutoInsertSpaces(QDebug* self, bool b) {
    self->setAutoInsertSpaces(b);
}

bool QDebug_QuoteStrings(const QDebug* self) {
    return self->quoteStrings();
}

void QDebug_SetQuoteStrings(QDebug* self, bool b) {
    self->setQuoteStrings(b);
}

QDebug* QDebug_Quote(QDebug* self) {
    QDebug& _ret = self->quote();
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_Noquote(QDebug* self) {
    QDebug& _ret = self->noquote();
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_MaybeQuote(QDebug* self) {
    QDebug& _ret = self->maybeQuote();
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft(QDebug* self, QChar* t) {
    QDebug& _ret = self->operator<<(*t);
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft2(QDebug* self, bool t) {
    QDebug& _ret = self->operator<<(t);
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft3(QDebug* self, char t) {
    QDebug& _ret = self->operator<<(static_cast<char>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft4(QDebug* self, int16_t t) {
    QDebug& _ret = self->operator<<(static_cast<short>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft5(QDebug* self, uint16_t t) {
    QDebug& _ret = self->operator<<(static_cast<unsigned short>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft8(QDebug* self, int t) {
    QDebug& _ret = self->operator<<(static_cast<int>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft9(QDebug* self, unsigned int t) {
    QDebug& _ret = self->operator<<(static_cast<unsigned int>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft10(QDebug* self, long t) {
    QDebug& _ret = self->operator<<(static_cast<long>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft11(QDebug* self, unsigned long t) {
    QDebug& _ret = self->operator<<(static_cast<unsigned long>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft12(QDebug* self, long long t) {
    QDebug& _ret = self->operator<<(static_cast<qint64>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft13(QDebug* self, unsigned long long t) {
    QDebug& _ret = self->operator<<(static_cast<quint64>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft15(QDebug* self, float t) {
    QDebug& _ret = self->operator<<(static_cast<float>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft16(QDebug* self, double t) {
    QDebug& _ret = self->operator<<(static_cast<double>(t));
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft17(QDebug* self, const char* t) {
    QDebug& _ret = self->operator<<(t);
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft19(QDebug* self, const libqt_string t) {
    QString t_QString = QString::fromUtf8(t.data, t.len);
    QDebug& _ret = self->operator<<(t_QString);
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft23(QDebug* self, const libqt_string t) {
    QByteArray t_QByteArray(t.data, t.len);
    QDebug& _ret = self->operator<<(t_QByteArray);
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft24(QDebug* self, QByteArrayView* t) {
    QDebug& _ret = self->operator<<(*t);
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_OperatorShiftLeft25(QDebug* self, const void* t) {
    QDebug& _ret = self->operator<<(t);
    // Cast returned reference into pointer
    return &_ret;
}

QDebug* QDebug_MaybeQuote1(QDebug* self, char c) {
    QDebug& _ret = self->maybeQuote(static_cast<char>(c));
    // Cast returned reference into pointer
    return &_ret;
}

void QDebug_Delete(QDebug* self) {
    delete self;
}

QDebugStateSaver* QDebugStateSaver_new(QDebug* dbg) {
    return new QDebugStateSaver(*dbg);
}

void QDebugStateSaver_Delete(QDebugStateSaver* self) {
    delete self;
}

QNoDebug* QNoDebug_new(const QNoDebug* other) {
    return new QNoDebug(*other);
}

QNoDebug* QNoDebug_new2(QNoDebug* other) {
    return new QNoDebug(std::move(*other));
}

void QNoDebug_CopyAssign(QNoDebug* self, QNoDebug* other) {
    *self = *other;
}

void QNoDebug_MoveAssign(QNoDebug* self, QNoDebug* other) {
    *self = std::move(*other);
}

QNoDebug* QNoDebug_Space(QNoDebug* self) {
    QNoDebug& _ret = self->space();
    // Cast returned reference into pointer
    return &_ret;
}

QNoDebug* QNoDebug_Nospace(QNoDebug* self) {
    QNoDebug& _ret = self->nospace();
    // Cast returned reference into pointer
    return &_ret;
}

QNoDebug* QNoDebug_MaybeSpace(QNoDebug* self) {
    QNoDebug& _ret = self->maybeSpace();
    // Cast returned reference into pointer
    return &_ret;
}

QNoDebug* QNoDebug_Quote(QNoDebug* self) {
    QNoDebug& _ret = self->quote();
    // Cast returned reference into pointer
    return &_ret;
}

QNoDebug* QNoDebug_Noquote(QNoDebug* self) {
    QNoDebug& _ret = self->noquote();
    // Cast returned reference into pointer
    return &_ret;
}

QNoDebug* QNoDebug_MaybeQuote(QNoDebug* self) {
    QNoDebug& _ret = self->maybeQuote();
    // Cast returned reference into pointer
    return &_ret;
}

QNoDebug* QNoDebug_Verbosity(QNoDebug* self, int param1) {
    QNoDebug& _ret = self->verbosity(static_cast<int>(param1));
    // Cast returned reference into pointer
    return &_ret;
}

QNoDebug* QNoDebug_MaybeQuote1(QNoDebug* self, const char param1) {
    QNoDebug& _ret = self->maybeQuote(static_cast<const char>(param1));
    // Cast returned reference into pointer
    return &_ret;
}

void QNoDebug_Delete(QNoDebug* self) {
    delete self;
}
